### DISCUSSION TITLE
Applications Going Offline Intermittently

### DISCUSSION THREAD

**Discussion by jokinvipo** [engageub/InternetIncome#113] [Q&A]
Hello,

I have noticed an intermittent issue where some containers for the same application on my dedicated server go offline after a short period (around 30 minutes to 1 hour) following a server restart. Despite the server restart resolving the issue temporarily, the containers continue to go offline unpredictably.

Server Specifications:
CPU: Ryzen 7 1700
RAM: 64GB

Observations:
Proxies are functioning correctly.
CPU usage remains at 30-50%.
RAM usage remains 20-30%
Delaying the startup of instances does not consistently resolve the issue.
Restarting the instance multiple times always works as a temporary fix.

I would appreciate any insights or suggestions on how to diagnose and permanently resolve this issue. Has anyone else experienced similar problems with containers going offline intermittently? Any advice or recommendations would be highly appreciated.

**Comment by engageub** [comment]
Hello,
Adnade and ebesucher go offline every hour as they are restarted but rest of the applications are not restarted through script.
Could you please be specific what applications are going offline.
If all the applications are going offline then there is something wrong at your end.

Thank you

**Reply by engageub** [reply]
You may also run the following command to see what containers are being restarted. 

```
sudo docker container ls
```

Thank you

**Reply by jokinvipo** [reply]
All earnapp and honeygain applications are offline the most time:

at the moment im not using Adnade and ebesucher yet, only the other ones


> ![image](https://github.com/engageub/InternetIncome/assets/35331883/8877e7d4-7b54-4313-9d5b-c0c4c2a7ac7d)

**Comment by engageub** [comment]
The screenshot shows that the script was started 5 hours ago but they were restarted an hour back.
Did you restart your PC an hour ago?

Thank you

**Reply by jokinvipo** [reply]
yes i made a server reboot to fix temporarly

thanks for the help

**Reply by engageub** [reply]
For earnapp, try to use only 10 to 15 devices per account.
The script is an integration of multiple docker images and I do own the  docker images.
This happened to me with earnapp earlier but on restarting the container, it was online. 
They usually come back online automatically but in case they don't you may use the following command. 

```
sudo docker restart <container_name> 
```
There are various reasons why this can happen. But I don't have control over issues arising due to network outage. 
The applications are designed that way.

Thank you

**Reply by engageub** [reply]
If you are using main branch for honeygain, could you please share the screenshot.
Some of the users were having trouble with proxies while using honeygain. Device was showing connected but there was no traffic.
Could you please let me know if you are seeing traffic in honeygain dashboard.

Thank you

**Reply by jokinvipo** [reply]
i have this structure bellow:

one folder with main and all APs (excluding Adnade and ebesucher):

InternetIncome-main , HoneyGain01, HoneyGain02, xx... 05

and i made this change in script as well: internetIncome.sh honeygain/honeygain:0.6.6

**Reply by engageub** [reply]
Thank you for the confirmation regarding honeygain.
On your PC, if only honeygain and earnapp are having issues, it might be a problem with the container or the network.
If all the applications are not working with proxy, that's a different issue.
There can be a possibility that your network is throttling DNS requests through UDP. 
Can you enable DNS_OVER_HTTPS in properties.conf and start the containers.
If your are using socks5 proxies and they support UDP, please use socks5 DNS.

Thank you

**Reply by jokinvipo** [reply]
Thanks alot for the help!

Im going to try this DNS_OVER_HTTPS to check if there is any change.

I saw in your main page that per IP you can make +/-50usd per month. Can you tell me wich apps are worth to use atm ?

**Reply by engageub** [reply]
Hello,
If you are using oracle free VPS, you can install the apps mentioned in the table which support VPS.
The income is based on the app, IP quality and demand in the region.
Usually, US,UK  region IPs, pay better than others. 

Thank you

**Reply by engageub** [reply]
Hello,
A restart script has been implemented for earnapp as most of the nodes are going offline once in three days.
This daily restart script will keep the nodes online and earnings are consistent.
Changes are pushed to test branch. 

https://github.com/engageub/InternetIncome/tree/test

Thank you

**Comment by engageub** [comment]
Hi, 
Please read the [FAQ](https://github.com/engageub/InternetIncome?tab=readme-ov-file#-why-are-some-of-the-containers-for-the-same-application-offline) which has been recently updated. 


If you see issues related to DNS queries, please use [test](https://github.com/engageub/InternetIncome/tree/test) branch with DNS over HTTPS enabled.

Thank you
